src/collections/options.ts

```diff
@@ -138,9 +138,17 @@ export const updateOneInternalOptionsKeys: Set<string> = new Set(
  * CreateCollectionOptions
  */
 
-export interface CreateCollectionOptions {
-  vectors?: {
-    size: number,
-    function?: 'cosine' | 'euclidean' | 'dot_product'
-  }
-}
\ No newline at end of file
+class _VectorOptions {
+    size: number = 0;
+    function?: 'cosine' | 'euclidean' | 'dot_product';
+}
+
+export class _CreateCollectionOptions {
+    vector?: _VectorOptions = undefined;
```

Comment:
I'd prefer to avoid adding a separate `_VectorOptions` class, avoid seeing `_VectorOptions` pop up in autocomplete. The following should work a little better.

```
class _CreateCollectionOptions {
    vector?: {
        size: number,
        function?: 'cosine' | 'euclidean' | 'dot_product'
    } = undefined
}
```